### PR TITLE
feat: add a training loss calculation to the predict method of PLT reduction

### PR DESCRIPTION
**Pull request by mwydmuch**
This is a small change to PLT reduction that adds calculation of training loss also to the `predict` method when labels are available (e.g., when predicting for holdout dataset), instead of always returning 0. Which is a behavior that might cause confusion/problems to some users (as in #4511).

### CODE REVIEW

File: vowpalwabbit/core/src/reductions/plt.cc [hunk 1]
```diff
@@ -139,6 +132,18 @@ void learn(plt& p, learner& base, VW::example& ec)
...
+}
+
+void learn(plt& p, learner& base, VW::example& ec)
+{
+  auto multilabels = std::move(ec.l.multilabels);
+  VW::polyprediction pred = std::move(ec.pred);
+
+  double t = p.all->sd->t;
+  double weighted_holdout_examples = p.all->sd->weighted_holdout_examples;
+  p.all->sd->weighted_holdout_examples = 0;
```
Comment (jackgerrits) [inline vowpalwabbit/core/src/reductions/plt.cc:144]: Why reset this to zero?

Comment (mwydmuch) [reply vowpalwabbit/core/src/reductions/plt.cc:144]: Good question. This is there from the initial implementation, it is not a change I introduced in this PR, but I don't remember why it's here. Does it somehow impact the update of the base classifier? That probably could be a reason for "resetting" this variable. If not, this can probably be removed.

Comment (mwydmuch) [reply vowpalwabbit/core/src/reductions/plt.cc:144]: Ok, it seems that it doesn't impact the training so I think I will remove these.

Comment (jackgerrits) [reply vowpalwabbit/core/src/reductions/plt.cc:144]: Sounds good!

**Review by jackgerrits — APPROVED** [review]
Thanks for this change! The loss being reported now is really helpful.

File: vowpalwabbit/core/src/reductions/plt.cc [hunk 2]
```diff
@@ -154,8 +151,6 @@ void learn(plt& p, learner& base, VW::example& ec)
   for (auto& n : p.negative_nodes) { loss += learn_node(p, n, base, ec); }
 
   p.all->sd->t = t;
```
Comment (jackgerrits) [inline vowpalwabbit/core/src/reductions/plt.cc:153]: Can you avoid setting t here?

**Comment by mwydmuch** [comment]
Ok, all the checks have passed, so from my point, this is ready to be merged :)
